Kpi review2

jsieland opened the pull request:
Hi everyone,

I finally pushed the KPI review ;)
As always: feedback is much appreciated!

### What I did
For the most part it was an editorial review and every KPI should have at least one example now (except for title).
But the following KPI got some mentionable updates.

Title
* allow brackets (for abbreviations, see sidequests below in addition)

Description:
* re-structured the guidance part for better readability and to avoid duplicating basically the same things in different words
* added example

Distribution Providers
* update according to latest changes in "contacts"
*  removed "organizations" because this is mandatory in WCMP2
*  included contact instructions

Links health
* added MQTTS (including examples)


### Up for debate

1. KPI for valid `channel`

In https://github.com/wmo-im/wcmp2/issues/79#issuecomment-1717155519 is mentioned that `channel` should be valid. But in WCMP2 there is REQ-15c: 

> A WCMP record links property **SHALL contain the MQTT topic information** for real-time data under which the data publication notifications will be accessible from the WIS2 Global Broker, **following the WIS2 topic hierarchy**.

As I understand this requirement, there is some check that the TH (used for `channel`) is valid. Which means there is no need for a KPI and (in my opinion) we should give an additional reference to https://wmo-im.github.io/wis2-topic-hierarchy/standard/wis2-topic-hierarchy-DRAFT.html#_publishing_guidelines for further information. 
But is that really the case or is this done by infrastructure as discussed in our last meeting (2023-10-23)? I think we really need to clarify this because it helps users understand these relations.

2. links health: check for `rel=collection` or `rel=items/rel=item`

Also mentioned in https://github.com/wmo-im/wcmp2/issues/79#issuecomment-1717155519 but I can not remember the reason(s) for this one... also this bullet point is listed below "ACTION: add to WIS2 Guide" so not sure where to put a possible KPI? @tomkralidis or @amilan17 can you remember how we came up with that?


### Sidequests for pywcmp
As @antje-s and I updated our examples we got downgraded due to abbreviations in title and description but the KPI-title allows for up to 3 abbreviations. Also in guidance for KPI-description it is recommended to "Spell out uncommon acronyms only once" and "Avoid spelling out commonly used acronym which are already understood by the general public" - so for this part pywcmp should be checked and adjusted accordingly.

tomkralidis commented:
Thanks @jsieland !  Comments:

1. KPI for valid channel: yes, [Requirement 15C](https://wmo-im.github.io/wcmp2/standard/wcmp2-DRAFT.html#_overview_3) negates the need for a KPI per se. Side note: from an ETS perspective, we will check for a link `channel`; if found, will validate (given it is difficult to technically decipher real-time vs other data)
2. links health: check for `rel=collection` or `rel=items/rel=item` : good question.  Unless we have detailed notes, we can ignore  for now on this addition
3. From the title KPI, can we change the test for "Title Case" ("My Dataset") to sentence case ("My dataset")?

For pywcmp, we will update once this PR is approved/merged.

jsieland commented:
Thanks @tomkralidis !

>     3. From the title KPI, can we change the test for "Title Case" ("My Dataset") to sentence case ("My dataset")?

Sure. I think it's important that we have a consistent way but in my opinion it doesn't matter if title or sentence case. I'll update the KPI.


Another side note I just realised while chatting with @antje-s: 

WCMP2 REQ-12c states that if a role is given, it should be one from https://github.com/wmo-im/wcmp2-codelists/blob/main/codelists/contact-role.csv which contains no "distributor" ... BUT we have a KPI "Distribution Providers" ... so either we add this role to the codelist or we need to rewrite the KPI (maybe for host?) or we forget about this KPI. Sorry for missing this earlier!

Any opinions on that?

tomkralidis commented:
> WCMP2 REQ-12c states that if a role is given, it should be one from https://github.com/wmo-im/wcmp2-codelists/blob/main/codelists/contact-role.csv which contains no "distributor" ... BUT we have a KPI "Distribution Providers" ... so either we add this role to the codelist or we need to rewrite the KPI (maybe for host?) or we forget about this KPI. Sorry for missing this earlier!
>
> Any opinions on that?

Good catch!  I think we should update the KPI to look for `host` (and update the JSON snippets inline with the latest WCMP2/OARec.

jsieland commented:
@tomkralidis 

I updated the PR with the following:
- KPI for title: changed from title case to sentence case
- KPI for distribution provider: rewrite for host and renamed KPI for alignment with WCMP2 spec

tomkralidis commented:
Excellent work again @jsieland! Merging.
